### DISCUSSION TITLE
new unseen combination of labels

### DISCUSSION THREAD

**Discussion by yishairasowsky** [scikit-multilearn/scikit-multilearn#280] [Q&A] [answered]
@niedakh @queirozfcom @fmaguire @Antiavanti @elzbietaZ 
Hi! And thank you for your awesome service! I use your product heavily, with very helpful and insightful results! Much appreciated! Maybe you can help me understand in labelpowerset what happens when there is a combination of labels in the test set which was unseen, i.e. it did not exist in the training set. How can the algo possibly predict the right answer? Maybe there is a solution for this. Thank you!
yishairasowsky@gmail.com

**Comment by yishairasowsky** [comment]
@ChristianSch @ljvmiranda921 @queirozfcom @fbenites @Antiavanti @grzegorz700 @Jopepato @ameet-1997 @chrysm @grzegorz700 @niedakh 
Does anyone have a minute to perhaps answer for this?

**Comment by ameet-1997** [comment]
The label combinations are computed on the training set, so all the possible predictions will be some combination of labels seen during train time.

So the short answer is that unseen label combinations will never be predicted, and thus the model will always get the wrong.

This, and the fact that there are exponential possible label combinations which makes the computation expensive, are the reasons that label powerset does not scale well with increase in number of labels.

**Comment by fbenites** [comment, marked as answer]
There are some algorithms based on labelpowersets which can cope with unseen labels in the multi-label classification (HOMER, Rakel). Also ARAM can cope with that since it makes voting over multiple labels and uses a threshold. Normally multi-label problems are relatively sparse, so that labelpowersets can compress the number of ensemble classifiers, again this depends heavily on the cardinality. Also other method to cope with the correlation would be ensemble of classifier chains, but is very computationally expensive. So using ensemble is anyone a way to cope with the problem, especially using bootstrapping (so subsets of the labelpowerset as in Rakel), and using e.g. voting to create an whole labelset again. HEre you can also have false correlations, i.e. some you might never had in the training data, but which might contradict the ontology. There is no free lunch here too...

**Comment by yishairasowsky** [comment]
thanks for your reply!
